src/Firebase/IdentityPlatform.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kreait\Firebase;
+
+use Kreait\Firebase\IdentityPlatform\ApiClient;
+use Kreait\Firebase\IdentityPlatform\DefaultSupportedIdpConfig;
+use Kreait\Firebase\IdentityPlatform\InboundSamlConfig;
+use Kreait\Firebase\IdentityPlatform\OAuthIdpConfig;
+use Kreait\Firebase\Request\DefaultSupportedIdpConfig as DefaultSupportedIdpConfigRequest;
+use Kreait\Firebase\Request\InboundSamlConfig as InboundSamlConfigRequest;
+use Kreait\Firebase\Request\OAuthIdpConfig as OAuthIdpConfigRequest;
+use Kreait\Firebase\Util\ArrayHelper;
+use Kreait\Firebase\Util\JSON;
+use Psr\Http\Message\ResponseInterface;
+
+class IdentityPlatform implements Contract\IdentityPlatform
+{
+    private ApiClient $client;
+
+    /**
+     * @param iterable<ApiClient|null>|ApiClient|null ...$x
+     *
+     * @internal
+     */
+    public function __construct(...$x)
+    {
+        foreach ($x as $arg) {
+            if ($arg instanceof ApiClient) {
+                $this->client = $arg;
+            }
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function listAllDefaultSupportedIdpConfigs(): array
+    {
+        $response = $this->client->listAllDefaultSupportedIdpConfigs();
+
+        return ArrayHelper::flatten($this->getResponseAsArray($response));
```

Comment:
Would it be possible to integrate the flattening into the class and remove the `ArrayHelper`? I believe I've seen it used only here… even if not, I don't mind code duplication here and there. (I know, that's a bold statement considering I introduced the `DT` helper back in the time, but please bear with me 😅)